mmdeploy/codebase/mmflow/deploy/flow.py

```diff
@@ -0,0 +1,309 @@
+        if not isinstance(result[0], np.ndarray):
+            result = [result[0].detach().cpu().numpy()]
+        return result
+
+    @staticmethod
+    def get_partition_cfg(partition_type: str, **kwargs) -> Dict:
+        """Get a certain partition config for mmedit.
+
+        Args:
+            partition_type (str): A string specifying partition type.
+
+        Returns:
+            dict: A dictionary of partition config.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def get_tensor_from_input(input_data: Dict[str, Any]) -> torch.Tensor:
+        """Get input tensor from input data.
+
+        Args:
+            input_data (dict): Input data containing meta info
+            and image tensor.
+        Returns:
+            torch.Tensor: An image in `Tensor`.
+        """
+        return input_data['lq']
+
+    @staticmethod
+    def evaluate_outputs(model_cfg,
+                         outputs: list,
+                         dataset: Dataset,
+                         metrics: Optional[str] = None,
+                         out: Optional[str] = None,
+                         metric_options: Optional[dict] = None,
+                         format_only: bool = False,
+                         log_file: Optional[str] = None,
+                         json_file: Optional[str] = None,
+                         **kwargs) -> None:
+        """Evaluation function implemented in mmflow.
+
+        Args:
+            model_cfg (mmcv.Config): The model config.
+            outputs (list): A list of result of model inference.
+            dataset (Dataset): Input dataset to run test.
+            metrics (str): Evaluation metrics, which depends on
+                the codebase and the dataset, e.g., "EPE", "Fl-all" in mmflow.
+            out (str): Output result file in pickle format, defaults to `None`.
+            metric_options (dict): Custom options for evaluation, will be
+                kwargs for dataset.evaluate() function. Defaults to `None`.
+            format_only (bool): Format the output results without perform
+                evaluation. It is useful when you want to format the result
+                to a specific format and submit it to the test server. Defaults
+                to `False`.
+            log_file (str | None): The file to write the evaluation results.
+                Defaults to `None` and the results will only print on stdout.
+        """
+        from mmcv.utils import get_logger
+        logger = get_logger('test', log_file=log_file)
+        if out:
+            logger.debug(f'writing results to {out}')
+            mmcv.dump(outputs, out)
+
+        # if json_file is not None:
+        #     mmcv.dump(stats, json_file, indent=4)
+
+    def get_preprocess(self) -> Dict:
+        """Get the preprocess information for SDK.
+
+        Return:
+            dict: Composed of the preprocess information.
+        """
+        input_shape = get_input_shape(self.deploy_cfg)
```

Comment:
does mmflow have special pre-post processes? If they are not supported in sdk. then we could directly raise not implemented error.